scene/gui/scroll_container.cpp

```diff
@@ -434,41 +434,37 @@ void ScrollContainer::update_scrollbars() {
 
 	Size2 min = child_max_size;
 
-	bool hide_scroll_v = !scroll_v || min.height <= size.height;
-	bool hide_scroll_h = !scroll_h || min.width <= size.width;
-
-	v_scroll->set_max(min.height);
-	if (hide_scroll_v) {
-		v_scroll->set_page(size.height);
-		v_scroll->hide();
-		scroll.y = 0;
-	} else {
-		v_scroll->show();
-		if (hide_scroll_h) {
-			v_scroll->set_page(size.height);
-		} else {
-			v_scroll->set_page(size.height - hmin.height);
-		}
-
-		scroll.y = v_scroll->get_value();
-	}
+	bool hide_scroll_h = !scroll_h || min.width <= size.width || !h_scroll_visible;
+	bool hide_scroll_v = !scroll_v || min.height <= size.height || !v_scroll_visible;
 
 	h_scroll->set_max(min.width);
+	h_scroll->set_page(size.width - (hide_scroll_v ? 0 : vmin.width));
 	if (hide_scroll_h) {
-		h_scroll->set_page(size.width);
 		h_scroll->hide();
-		scroll.x = 0;
+
+		if (h_scroll_visible) {
+			scroll.x = 0; // don't reset when setting h_scroll_visible to false
+		}
```

Comment:
I tested with this part removed and I don't see any difference. What does it address exactly?

Reply:
The `scroll.x = 0;` was there originally, the if statement I added so that it wouldn't set the scroll back to 0 when unchecking `h_scroll_visible`, but looking at it it looks like the scroll variable is pretty much redundant, so I just removed it